### PR TITLE
Fix bug cannot view email node configuration in campaign builder without view others email permission

### PR DESCRIPTION
**Pull request by vln104**
# Reproduction:
1) Create a role with the following email permission: view, edit, publish delete own and create. Note that view others permission is not granted. Also grant permission to view and edit campaigns.
2) Create user with above role
3) Assume a previous campaign with send email node is available, otherwise create one.
4) Login as above user, go to campaigns builder and try to open the send email configuration node.
5) Notice that the configuration box does not load

The log below is observed in the log file:
[2015-11-17 09:26:35] mautic.ERROR: Fatal: Call to a member function getId() on a non-object - in file /homepages/6/d416965206/htdocs/rightmarket/poc-mautic/app/bundles/EmailBundle/Entity/EmailRepository.php - at line 249 [] 

This PR fixes the issue by setting the current user object.

Also just a note for people testing this, number 1148(#1148) is also related so might be worth testing it while you are here as well.

### CODE REVIEW

**Comment by escopecz** [comment]
Thanks for the PR! I'd add to the 1) to also create permission to view and edit campaigns.

The error message is gone, but for me, another one appear:
```
500 Internal Server Error - [Semantical Error] line 0, col 103 near 'createdBy) =': Error: Invalid PathExpression. Must be a SingleValuedAssociationField. 

vendor/doctrine/orm/lib/Doctrine/ORM/Query/QueryException.php (line #63):
vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php (line #483): at Doctrine\ORM\Query\QueryException :: semanticalError ( 'line 0, col 103 near 'createdBy) =': Error: Invalid PathExpression. Must be a SingleValuedAssociationField.', object(QueryException) )
vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php (line #800): at Doctrine\ORM\Query\Parser -> semanticalError ( 'Invalid PathExpression. Must be a SingleValuedAssociationField.', array('value' => 'createdBy', 'type' => '100', 'position' => '103') )
vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php (line #279): at Doctrine\ORM\Query\Parser -> processDeferredPathExpressions ( object(SelectStatement) )
vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php (line #351): at Doctrine\ORM\Query\Parser -> getAST ( )
vendor/doctrine/orm/lib/Doctrine/ORM/Query.php (line #255): at Doctrine\ORM\Query\Parser -> parse ( )
vendor/doctrine/orm/lib/Doctrine/ORM/Query.php (line #267): at Doctrine\ORM\Query -> _parse ( )
vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php (line #794): at Doctrine\ORM\Query -> _doExecute ( )
vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php (line #586): at Doctrine\ORM\AbstractQuery -> execute ( null, '2' )
app/bundles/EmailBundle/Entity/EmailRepository.php (line #269): at Doctrine\ORM\AbstractQuery -> getArrayResult ( )
app/bundles/EmailBundle/Form/Type/EmailListType.php (line #56): at Mautic\EmailBundle\Entity\EmailRepository -> getEmailList ( '', '0', '0', false, true, 'template' )
vendor/symfony/options-resolver/Symfony/Component/OptionsResolver/OptionsResolver.php (line #843): at Mautic\EmailBundle\Form\Type\EmailListType -> Mautic\EmailBundle\Form\Type\{closure} ( object(OptionsResolver), array() )
app/bundles/EmailBundle/Form/Type/EmailListType.php (line #74): at Symfony\Component\OptionsResolver\OptionsResolver -> offsetGet ( 'choices' )
vendor/symfony/options-resolver/Symfony/Component/OptionsResolver/OptionsResolver.php (line #843): at Mautic\EmailBundle\Form\Type\EmailListType -> Mautic\EmailBundle\Form\Type\{closure} ( object(OptionsResolver), false )
vendor/symfony/options-resolver/Symfony/Component/OptionsResolver/OptionsResolver.php (line #776): at Symfony\Component\OptionsResolver\OptionsResolver -> offsetGet ( 'disabled' )
vendor/symfony/form/Symfony/Component/Form/ResolvedFormType.php (line #104): at Symfony\Component\OptionsResolver\OptionsResolver -> resolve ( array('label' => 'mautic.email.send.selectemails', 'label_attr' => array('class' => 'control-label'), 'attr' => array('class' => 'form-control', 'tooltip' => 'mautic.email.choose.emails_descr'), 'multiple' => false, 'constraints' => array(object(NotBlank))) )
vendor/symfony/form/Symfony/Component/Form/Extension/DataCollector/Proxy/ResolvedTypeDataCollectorProxy.php (line #82): at Symfony\Component\Form\ResolvedFormType -> createBuilder ( object(FormFactory), 'email', array('label' => 'mautic.email.send.selectemails', 'label_attr' => array('class' => 'control-label'), 'attr' => array('class' => 'form-control', 'tooltip' => 'mautic.email.choose.emails_descr'), 'multiple' => false, 'constraints' => array(object(NotBlank))) )
vendor/symfony/form/Symfony/Component/Form/FormFactory.php (line #87): at Symfony\Component\Form\Extension\DataCollector\Proxy\ResolvedTypeDataCollectorProxy -> createBuilder ( object(FormFactory), 'email', array('label' => 'mautic.email.send.selectemails', 'label_attr' => array('class' => 'control-label'), 'attr' => array('class' => 'form-control', 'tooltip' => 'mautic.email.choose.emails_descr'), 'multiple' => false, 'constraints' => array(object(NotBlank))) )
vendor/symfony/form/Symfony/Component/Form/FormBuilder.php (line #106): at Symfony\Component\Form\FormFactory -> createNamedBuilder ( 'email', 'email_list', null, array('label' => 'mautic.email.send.selectemails', 'label_attr' => array('class' => 'control-label'), 'attr' => array('class' => 'form-control', 'tooltip' => 'mautic.email.choose.emails_descr'), 'multiple' => false, 'constraints' => array(object(NotBlank))) )
vendor/symfony/form/Symfony/Component/Form/FormBuilder.php (line #267): at Symfony\Component\Form\FormBuilder -> create ( 'email', 'email_list', array('label' => 'mautic.email.send.selectemails', 'label_attr' => array('class' => 'control-label'), 'attr' => array('class' => 'form-control', 'tooltip' => 'mautic.email.choose.emails_descr'), 'multiple' => false, 'constraints' => array(object(NotBlank))) )
vendor/symfony/form/Symfony/Component/Form/FormBuilder.php (line #215): at Symfony\Component\Form\FormBuilder -> resolveChildren ( )
vendor/symfony/form/Symfony/Component/Form/FormBuilder.php (line #221): at Symfony\Component\Form\FormBuilder -> getForm ( )
vendor/symfony/form/Symfony/Component/Form/FormFactory.php (line #39): at Symfony\Component\Form\FormBuilder -> getForm ( )
app/bundles/CampaignBundle/Controller/EventController.php (line #212): at Symfony\Component\Form\FormFactory -> create ( 'campaignevent', array('id' => '29', 'name' => 'Email o dalším postupu', 'description' => null, 'type' => 'email.send', 'eventType' => 'action', 'order' => '1', 'properties' => array('email' => '42'), 'triggerDate' => null, 'triggerInterval' => '1', 'triggerIntervalUnit' => 'd', 'triggerMode' => 'immediate', 'decisionPath' => null, 'tempId' => 'newac4c60fc424210353c153f7b146745e66e7d3d1f', 'changes' => null), array('action' => '/mautic-github/index_dev.php/s/campaigns/events/edit/29', 'settings' => array('label' => 'Send email', 'description' => 'Send the selected email to the lead.', 'callback' => array('\Mautic\EmailBundle\Helper\CampaignEventHelper', 'sendEmailAction'), 'formType' => 'emailsend_list', 'formTypeOptions' => array('update_select' => 'campaignevent_properties_email'), 'formTheme' => 'MauticEmailBundle:FormTheme\EmailSendList')) )
app/bundles/CoreBundle/Controller/CommonController.php (line #308): at Mautic\CampaignBundle\Controller\EventController -> editAction ( '29', '' )
at Mautic\CoreBundle\Controller\CommonController -> executeAction ( 'edit', '29', '0', '' )
app/bootstrap.php.cache (line #3051): at call_user_func_array ( array(object(EventController), 'executeAction'), array('edit', '29', '0', '') )
app/bootstrap.php.cache (line #3013): at Symfony\Component\HttpKernel\HttpKernel -> handleRaw ( object(Request), '1' )
app/bootstrap.php.cache (line #3162): at Symfony\Component\HttpKernel\HttpKernel -> handle ( object(Request), '1', true )
app/bootstrap.php.cache (line #2403): at Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel -> handle ( object(Request), '1', true )
app/AppKernel.php (line #112): at Symfony\Component\HttpKernel\Kernel -> handle ( object(Request), '1', true )
index_dev.php (line #61): at AppKernel -> handle ( object(Request) )

Previous Exceptions
SELECT partial e.{id, subject, name, language} FROM Mautic\EmailBundle\Entity\Email e WHERE IDENTITY(e.createdBy) = :id AND e.variantParent IS NULL AND e.emailType = 'template' ORDER BY e.name ASC
```
Tested on postgreSql.

**Comment by vln104** [comment]
Yes I can reproduce the issue - looking at it now.

(*and the standard developer it works on my machine*)

**Comment by vln104** [comment]
I've made a further change and I think it's working for me now - can open the email node configuration dialog

**Comment by escopecz** [comment]
I don't get any error in the Campaign Builder now. But I also cannot select any email there since I don't have rights to see other emails. That's correct. But I went ahead and test if my email would be selectable for me if I create it. But when I click New in the Email manager, I get this error:

```
500 Internal Server Error - Error: Call to a member function getId() on null
app/bundles/FormBundle/Entity/FormRepository.php (line #62):
```

Could you add fix for this one as well?

**Comment by vln104** [comment]
John, I think the issue you describe is in number 1148. I've considered it a separate issue so created a different PR. Please confirm. Thanks

**Comment by escopecz** [comment]
Yes, you are right. The error is fixed there. Could you fix the indentation again?

**Comment by vln104** [comment]
done

**Comment by escopecz** [comment]
I vote up :+1:

File: app/bundles/FormBundle/Model/FormModel.php [hunk 1]
```diff
@@ -490,7 +490,6 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml)
 
                 switch ($f->getType()) {
                     case 'text':
-                    case 'email':
```
Comment (alanhartless) [inline app/bundles/FormBundle/Model/FormModel.php]: Why was this line removed?

**Comment by vln104** [comment]
@alanhartless it was a fix for another issue. I was not familiar with git so had fixes for two different bugs in one branch. The addition of 'case email:' line to FormModel was put into a different branch and I believed it has already been merged

**Comment by alanhartless** [comment]
Ah I see.  It was #1129.  I fear that when we merge this PR, it's going to revert that fix. It seems that Git is a bit confused for some reason.  You might need to try this

```
git checkout staging && git pull && git checkout FixBugCampaignViewWithoutEmailViewOthers && git merge --no-ff staging
```

Then try pushing the changes to see if it fixes that.  If not, you might have to manually find it in the FixBugCampaignViewWithoutEmailViewOthers branch, add it back and push to keep this PR from removing it.

**Comment by alanhartless** [comment]
Hmm...actually looks like it's already been removed from staging with the merge of #1148.  It's not in staging now https://github.com/mautic/mautic/blob/staging/app/bundles/FormBundle/Model/FormModel.php#L492. I think you'll have to add it back and just push it to this branch.

**Comment by vln104** [comment]
@alanhartless I've done as you suggested. Hopefully it is all OK and makes sense now.

**Comment by alanhartless** [comment]
@vln104 Thanks! Looks like it removed that line from this PR but it's still gone from staging and I half expected for it to show up in this PR as being added back.  Git does this sometimes; we've had whole merge PRs somehow get reverted.  I'll manually merge this to make sure the fix is applied.

**Comment by alanhartless** [comment]
Looks like we're good! Thanks again @vln104!

**Pull request merged**